app/src/test/java/uk/gov/mint/LoadHandlerTest.java

```diff
@@ -24,13 +31,29 @@ public void handle_addsTheHashAndThenSavesInTheDatabase() throws Exception {
         LoadHandler loadHandler = new LoadHandler(entriesUpdateDAO, logStream);
 
         String payload = "{\"key1\":\"value1\"}\n{\"key2\":\"value2\"}";
+
         String expectedHash1 = Digest.shasum("{\"key1\":\"value1\"}");
         String expectedHash2 = Digest.shasum("{\"key2\":\"value2\"}");
 
+        final String entry1 = "{\"entry\":{\"key1\":\"value1\"},\"hash\":\"" + expectedHash1 + "\"}";
+        final byte[] entry1Bytes = canonicalise(entry1);
+        final String entry2 = "{\"entry\":{\"key2\":\"value2\"},\"hash\":\"" + expectedHash2 + "\"}";
+        final byte[] entry2Bytes = canonicalise(entry2);
+
         loadHandler.handle(payload);
 
-        verify(entriesUpdateDAO).add(("{\"entry\":{\"key1\":\"value1\"},\"hash\":\"" + expectedHash1 + "\"}").getBytes());
-        verify(entriesUpdateDAO).add(("{\"entry\":{\"key2\":\"value2\"},\"hash\":\"" + expectedHash2 + "\"}").getBytes());
-        verify(logStream, times(2)).notifyOfNewEntries();
+        ArgumentCaptor<List> captor = ArgumentCaptor.forClass(List.class);
+        verify(entriesUpdateDAO, times(1)).add(captor.capture());
+        final List<byte[]> payloadArray = captor.getValue();
```

Comment:
you can avoid unchecked assignment warnings with the [@Captor annotation](http://stackoverflow.com/questions/5606541/how-to-capture-a-list-of-specific-type-with-mockito)